fission/src/ui/components/MainHUD.tsx

```diff
@@ -169,6 +175,13 @@ const MainHUD: React.FC = () => {
                         icon={<GiSteeringWheel />}
                         onClick={() => MirabufCachingService.RemoveAll()}
                     />
+                    <MainHUDButton
+                        value={"Edit Scoring Zones"}
+                        icon={<IoBasketball />}
+                        onClick={() => {
+                            openPanel("scoring-zones")
+                        }}
+                    />
```

Comment:
<img width="555" alt="Screenshot 2024-07-15 at 11 18 41 AM" src="https://github.com/user-attachments/assets/047c9386-8528-47f5-b4e3-554ae99f754b">

Ran into this error after I pressed this button with a field and robot spawned

Reply:
Just ran into that error on another branch. Solution:
change PreferenceSystem.ts line 98 from 
`let allFieldPrefs = this.getPreference<{ [key: string]: FieldPreferences }>(RobotPreferencesKey)`
to
`let allFieldPrefs = this.getPreference<{ [key: string]: FieldPreferences }>(FieldPreferencesKey)`